sgkit/io/vcf/vcf_reader.py

```diff
@@ -902,3 +928,55 @@ def count_variants(path: PathType, region: Optional[str] = None) -> int:
         if region is not None:
             vcf = vcf(region)
         return sum(1 for _ in region_filter(vcf, region))
+
+
+def zarr_array_sizes(input: PathType) -> Dict[str, Any]:
+    """Make a pass through a VCF/BCF file to determine sizes for storage in Zarr."""
+
+    with open_vcf(input) as vcf:
+
+        ploidy = -1
+        alt_alleles = 0
+
+        info = _get_vcf_field_defs(vcf, "INFO")
+        info_field_defs = {
+            key: {"Number": 1} for key in info.keys() if info[key]["Number"] == "."
+        }
+
+        format = _get_vcf_field_defs(vcf, "FORMAT")
+        format_field_defs = {
+            key: {"Number": 1} for key in format.keys() if format[key]["Number"] == "."
+        }
+
+        for variant in vcf:
+            for key, val in info_field_defs.items():
+                field_val = variant.INFO.get(key)
+                if field_val is not None:
+                    try:
+                        val["Number"] = max(val["Number"], len(field_val))
+                    except TypeError:
+                        pass  # single value
+
+            for key, val in format_field_defs.items():
+                field_val = variant.format(key)
+                if field_val is not None:
+                    val["Number"] = max(val["Number"], field_val.shape[-1])
+
+            try:
+                ploidy = max(ploidy, variant.genotype.ploidy)
+            except Exception:
```

Comment:
Does Cyvcf2 really throw a raw Exception? You'd expect it to be a TypeError or something a bit more specific. Probably worth putting a note here because I'd imagine linting tools will start warning about stuff like this at some point and it would be helpful for future devs to know the rationale

Reply:
Fixed - it throws an AttributeException.